### ISSUE TITLE
Support for Spring Boot 2.0

### ISSUE THREAD

**Issue by sazzer** [graphql-java/graphql-spring-boot#40]
I've just tried it out, and whilst this project works great with Spring 5.0.0 and Spring Boot 1.5.x, it fails when using Spring Boot 2.0.x. The "graphiql-spring-boot" project works fine, but the actual GraphQL handlers don't ever start.

**Comment by apottere** [comment]
There's a PR open for updating to support Spring Boot 2.0.0, (#36), but I'm waiting for it to have a non-milestone release first.

**Comment by sdeleuze** [comment]
Shouldn't Boot 2.0 support also include support for Spring WebFlux ?

**Comment by apottere** [comment]
What parts of webflux would the servlet need to support?

**Comment by sazzer** [comment]
It seems a nice idea if the entire graphql server could be non blocking,
but I'm not sure how feasible that is since the graph structure
necessitates some parts to be computed only after other parts have finished.

**Comment by sdeleuze** [comment]
@apottere I guess the integration would not be as a Servlet, see https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html#spring-webflux. For example functional endpoints allow programmatic registration of routes.

@sazzer I am not familiar with GraphQL, but even if a full non-blocking solution for the graph could be appealing, maybe you could just support a similar integration than with Servlet as a first step, just leveraging WebFlux infrastructure instead of Servlet one. Does GraphQL processing requires using blocking IO ?

**Comment by sazzer** [comment]
I've no idea if it *requires* blocking IO per-se. However, what you often
have is almost reentrant processing - child nodes can only be evaluated
after the parent nodes are completely loaded, and I'm not sure if that
would be a problem at all. All the webflux examples I've seen have been:
read from database, convert to output, stream to client, and this would end
up being that structure repeated several times for one request.

**Comment by kganger** [comment]
It would be really helpful to get at least a snapshot release with this support

**Comment by SamD** [comment]
I just started my first spring boot project and seeing this happen with 1.5.9 RELEASE as well, any suggestions?

**Comment by marceloverdijk** [comment]
Is the plan to maintain/release different versions for Boot 1.5 and 2.0?

In that case it might be interesting to deliver a release based on latest Boot 2.0 M7. It’s quite stable and already used in production 😉

I’m currently experimenting with Graphql in an existing Boot 2.0 M7 app and unfortunately I have to create all beans manually as the Graphql Boot starter does not support 2.0.

What is holding you back to provide a release with 2.0 support (as multiple people are asking)?

**Comment by heruan** [comment]
I've rebased the PR and updated to Spring Framework 5.0.2.RELEASE and Spring Boot 2.0.0.M7 which is the latest milestone.

I understand @apottere concerns about releasing with a milestone dependency, and I agree to wait.

*But* I see no reason why not to release a milestone version of this plugin as well, with compatibility with Spring Boot 2.

**Comment by apottere** [comment]
I can definitely release a snapshot/milestone version with these changes - I hadn't thought of that.

**Comment by apottere** [comment]
Version `4.0.0.M1` is syncing to central now!

**Comment by marceloverdijk** [comment]
Thx @apottere I will test it asap.

**Comment by marceloverdijk** [comment]
Maybe I should raise this in a separate issue but wouldn't it be better to change the `om.graphql-java:graphql-java-tools` dependency from compile only to compile?

**Comment by apottere** [comment]
Yeah, I just forgot to update versions.  I'm not sure about depending on graphql-java-tools as a compile dependency, since the starter can be used without it.  Right now the specific auto-configuration triggers when it sees those classes on the classpath.

**Comment by marceloverdijk** [comment]
Yes, you are right regarding `graphql-java-tools`. I thought it was a mandatory dependency, but indeed it is optional.

**Comment by marceloverdijk** [comment]
Spring Boot `2.0.0 GA` was released today. Probably good to upgrade this project as well and maybe make a new release?

**Comment by ieugen** [comment]
Anything we can do to help out with the release?

**Comment by marceloverdijk** [comment]
I guess we need @apottere ..

**Comment by geowarin** [comment]
Hello!

For those interested, I wrote [an application](https://github.com/geowarin/graphql-webflux) in kotlin with graphq-java which uses webflux directly (no servlet).

I hope it will be useful if you guys decide to support webflux in the future.

**Comment by apottere** [comment]
Sorry it took me so long to get to this, it's released in `4.0.0`
